infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/fsm/InLogEntrySyncState.java

```diff
@@ -148,8 +152,15 @@ public void onEntry(LogReplicationState from) {
                 fsm.getAckReader().markSnapshotSyncInfoCompleted();
             }
 
-            logEntrySyncFuture = fsm.getLogReplicationFSMWorkers().submit(() ->
-                    logEntrySender.send(transitionEventId));
+            if (from.getType() == LogReplicationStateType.IN_LOG_ENTRY_SYNC && !logEntrySender.isTaskActive()) {
+                logEntrySyncFuture = fsm.getLogReplicationFSMWorkers().schedule(() ->
+                        logEntrySender.send(transitionEventId), waitRetryRead, TimeUnit.SECONDS);
+                waitRetryRead = Math.min(waitRetryRead + 1, WAIT_RETRY_READ_MAX);
```

Comment:
Increasing the waitRetry on every invocation can lead to an overflow error at some point.  I suggest not incrementing it after it reaches the limit(WAIT_RETRY_MAX).

Reply:
Since we set `waitRetryRead` by taking the min of itself and `WAIT_RETRY_MAX` each time the waitRetry is updated, we will never exceed the max value by more than 1 during the calculation

Reply:
Also, I was thinking we should start with miliseconds. It was seen that by adding some 200ms, the CPU spikes were gone.
This number again needs to be tuned since it can potentially increase LR's latency.